src/main/java/com/openlattice/search/SearchApi.java

```diff
@@ -277,23 +294,21 @@ DataSearchResult executeAdvancedEntitySetDataQuery(
             @Body Set<UUID> entityKeyIds );
 
     /**
-     * Executes a search for all neighbors of multiple entities of the same entity set that are connected by an
-     * association
+     * Executes a search for all neighbors of multiple entities of multiple entity sets that are connected by an
+     * association.
      *
-     * @param entitySetId the entity set id of the entities
-     * @param filter      optional constraints on entityKeyIds to include and filters on src/dst/edge entity set ids
+     * @param filter optional constraints on entityKeyIds of entity sets to include and filters on src/dst/edge
+     *               entity set ids.
      * @return A map from each entity id to a list of objects containing information about the neighbors and
-     * associations of that entity
+     * associations of that entity.
      */
-    @POST( BASE + ENTITY_SET_ID_PATH + NEIGHBORS + ADVANCED )
-    Map<UUID, List<NeighborEntityDetails>> executeFilteredEntityNeighborSearch(
-            @Path( ENTITY_SET_ID ) UUID entitySetId,
-            @Body EntityNeighborsFilter filter );
+    @POST( BASE + NEIGHBORS + ADVANCED )
+    Map<UUID, List<NeighborEntityDetails>> executeFilteredEntityNeighborSearch( @Body EntityNeighborsFilter filter );
 
-    @POST( BASE + ENTITY_SET_ID_PATH + NEIGHBORS + ADVANCED + IDS )
-    Map<UUID, Map<UUID, SetMultimap<UUID, NeighborEntityIds>>> executeFilteredEntityNeighborIdsSearch(
-            @Path( ENTITY_SET_ID ) UUID entitySetId,
-            @Body EntityNeighborsFilter filter );
+    @POST( BASE + NEIGHBORS + ADVANCED + IDS )
+    Map<UUID, Map<UUID, Map<UUID, Set<NeighborEntityIds>>>> executeFilteredEntityNeighborIdsSearch(
```

Comment:
same question as above re whether the actual data returned here is now different than it used to be (including keys with an empty set of values)

Reply:
so far none of the SearchController... rehearsal tests broke bc of this, but will investigate and get back

Reply:
ok. I mostly worry with these types of changes that projects will break in subtle ways, like if they're currently using map.containsKey(entityKeyId) to determine whether an entity has neighbors, or something like that. So it would just be good to know which (if any) are changing their behavior, so that the frontend apps can be modified accordingly, or so that we can preserve current behavior if that's preferable.

Reply:
So when serializing maps vs multimaps, maps will keep empty values, multimaps wont, but that can be avoided by removing empty values from the map implementation. I will look into the old/new code, if it was even possible or is now possible to have empty values.

Reply:
So, it was never possible to have an empty value and it's also not now, so this should be perfectly fine.

Old version:
```
Map<UUID, Map<UUID, SetMultimap<UUID, NeighborEntityIds>>> neighbors = Maps.newHashMap();
graphService.getEdgesAndNeighborsForVerticesBulk( entitySetIds, filter ).forEach( edge -> {
    ...
    NeighborEntityIds neighborEntityIds = new NeighborEntityIds( edge.getEdge().getEntityKeyId(), neighborEntityDataKey.getEntityKeyId(), isSrc );
    ...
    neighbors.get( entityKeyId ).get( edge.getEdge().getEntitySetId() ).put( neighborEntityDataKey.getEntitySetId(), neighborEntityIds );
    ...
}
```

New version:
```
val neighbors = mutableMapOf<UUID, MutableMap<UUID, MutableMap<UUID, MutableSet<NeighborEntityIds>>>>()
graphService.getEdgesAndNeighborsForVerticesBulk(filter).forEach { edge ->
    ...
    val neighborEntityIds = NeighborEntityIds(edge.edge.entityKeyId, neighborEntityDataKey.entityKeyId, isSrc)
    ...
    neighbors.getValue(entityKeyId).getValue(edgeEsId).putIfAbsent(neighborEsId, mutableSetOf())
    neighbors.getValue(entityKeyId).getValue(edgeEsId).getValue(neighborEsId).add(neighborEntityIds)
    ...
}
```

Reply:
ah cool! yay!